optuna/study/_optimize.py

```diff
@@ -56,8 +56,11 @@ def _optimize(
     if not study._optimize_lock.acquire(False):
         raise RuntimeError("Nested invocation of `Study.optimize` method isn't allowed.")
 
-    # TODO(crcrpar): Make progress bar work when n_jobs != 1.
-    progress_bar = pbar_module._ProgressBar(show_progress_bar and n_jobs == 1, n_trials, timeout)
+    if show_progress_bar and timeout is not None and n_jobs != 1:
```

Comment:
When `n_trials` is not `None`, progress bar uses `n_trials` instead of `timeout`:
```suggestion
    if show_progress_bar and n_trials is None and timeout is not None and n_jobs != 1:
```

Reply:
Ok ,fixed.